src/FsAutoComplete/LspServers/AdaptiveFSharpLspServer.fs

```diff
@@ -59,28 +59,22 @@ type AdaptiveWorkspaceChosen =
   | Projs of amap<string<LocalPath>, DateTime>
   | NotChosen
 
-type LoadedProjectFile =
-  { ProjectOptions: Types.ProjectOptions
-    FSharpProjectOptions: FSharpProjectOptions
-    LanguageVersion: LanguageVersionShim }
 
-type LoadedScriptFile =
+[<CustomEquality; NoComparison>]
+type LoadedProject =
   { FSharpProjectOptions: FSharpProjectOptions
     LanguageVersion: LanguageVersionShim }
 
-type LoadedProject =
-  | LoadedProjectFile of LoadedProjectFile
-  | LoadedScriptFile of LoadedScriptFile
+  interface IEquatable<LoadedProject> with
+    member x.Equals(other) =
+      x.FSharpProjectOptions = other.FSharpProjectOptions
 
-  member x.FSharpProjectOptions =
-    match x with
-    | LoadedProjectFile p -> p.FSharpProjectOptions
-    | LoadedScriptFile p -> p.FSharpProjectOptions
+  override x.GetHashCode() = x.FSharpProjectOptions.GetHashCode()
 
-  member x.LanguageVersion =
-    match x with
-    | LoadedProjectFile p -> p.LanguageVersion
-    | LoadedScriptFile p -> p.LanguageVersion
+  override x.Equals(other) =
+    match other with
+    | :? LoadedProject as other -> (x :> IEquatable<_>).Equals other
+    | _ -> false
```

Comment:
We don't really need to keep `ProjectOptions` so condensing this type. Also makes it use `FSharpProjectOptions` as its equality.